pkg/util/util.go

```diff
@@ -76,7 +81,8 @@ func NamespaceOpenShiftObject(componentName string, applicationName string) (str
 	}
 
 	// Return the hyphenated namespaced name
-	return fmt.Sprintf("%s-%s", strings.Replace(componentName, "/", "-", -1), applicationName), nil
+
+	return TruncateString(fmt.Sprintf("%s-%s", strings.Replace(componentName, "/", "-", -1), applicationName), maxAllowedNamespacedStringLength), nil
```

Comment:
Just a thought, can we rather truncate both componentName and applicationName individually to probably under maxAllowedNamespacedStringLength/2 so that both of them get a fair share?
I understand the current implementation accommodates a case where component name is very small but a meaningful application name is a bit large but can we try to strike a balance b/w them?

Reply:
But the PR already looks good to me however :) So I am ignoring the above

Reply:
I am perfectly fine with your suggestion @anmolbabu, but I haven't followed such an approach since I don't have the context of why it was initially created that way :)

Reply:
I see that as a convention followed to name dc, supervisor volume component container etc.. not sure why :) 
In my opinion, I would think that, this is a way to indicate which app it belongs to.. but there are also separate labels to indicate both component and application names respectively
Perhaps @kadel or @cdrage or @surajnarwade can help us with the reason

Reply:
> I am perfectly fine with your suggestion @anmolbabu, but I haven't followed such an approach since I don't have the context of why it was initially created that way :)

I think I created auto-generated component and application names with consequences of them being individually longer than an assumed char limit(of 12 chars) but without probably being aware of the consequence of each of them together being longer than any specific char limit...
And may be it was not caught initially either ... So, the bug :)

Reply:
Yeah, I wouldn't have been caught easily :)

I personally think this PR should do for now since the names aren't very important - all operations are done via labels